### PR TITLE
Add Pirate Bastion, Bulwark

### PR DESCRIPTION
**Pull request by 1010todd**
----------------------
**Content (Artwork / Ship)**

bulwark i29 thumb gear@2x

## Summary
Add pirate Bastion with scraps welded on to it as armour. Intended to be used by more established pirate groups with something they want to protect like their hideouts or planets under their protection against authorities or raids from a rival groups. 

<details>
<summary>Stats Difference</summary>

<table>
<tr>
	<td>Stat
	<td>Bastion
	<td>Bulwark
<tr>
	<td>shields
	<td>7500
	<td>7000
<tr>
	<td>hull
	<td>4700
	<td>5600
<tr>
	<td>req crew
	<td>17
	<td>19
<tr>
	<td>bunks
	<td>40
	<td>31
<tr>
	<td>mass
	<td>580
	<td>881
<tr>
	<td>drag
	<td>10.3
	<td>13.4
<tr>
	<td>heat dissipation
	<td>.5
	<td>.35
<tr>
	<td>cargo
	<td>110
	<td>12
<tr>
	<td>outfit
	<td>470
	<td>521
<tr>
	<td>weapon
	<td>190
	<td>223
</table>

</details>

<details>
<summary>Images</summary>

pirate bulwark@2x

pirate bulwark@2x

bulwark park@2x

</details>

## Save File
This save file can be used to play through the new mission content:
x

Test Plugin:
[Bulwark test.zip](https://github.com/endless-sky/endless-sky/files/10421181/Bulwark.test.zip)

## Artwork Checklist
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [x] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: https://github.com/endless-sky/endless-sky-assets/pull/119
 - [x] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/343

### CODE REVIEW

File: data/human/ships.txt [hunk 1]
```diff
@@ -806,6 +806,69 @@ ship "Bulk Freighter"
...
+	turret 0 0.5 "Heavy Anti-Missile Turret"
+	leak "leak" 40 50
+	leak "flame" 40 80
+	leak "big leak" 70 30
+	explode "tiny explosion" 5
+	explode "small explosion" 15
+	explode "medium explosion" 30
+	explode "large explosion" 20
+	"final explode" "final explosion medium"
+	description `The Bulwark is basically what you get for welding a bunch of scraps on to a Bastion. A tough, slow brawler most often used by warlords to protect their properties from raids either by the authorities or their rivals. Due to lack of speed and cargo space these ships rarely participate in raiding merchant convoys outside their home system.`
```
Comment (McloughlinGuy) [inline data/human/ships.txt:868]:
```suggestion
	description `The Bulwark is basically what you get by welding a bunch of scrap metal on to a Bastion. It is a tough, slow brawler that is most often used by warlords to protect their fleets from raids by either the authorities or their rivals. Due to their lack of speed and cargo space these ships rarely participate in raiding merchant convoys outside their home system.`

**Comment by McloughlinGuy** [comment]
Why does it have less crew than the bastion? How would welding scrap metal onto it make it need less crew members?

**Comment by Hurleveur** [comment]
Stats look quite weak to me, and I think we could be overusing buffing weapon (and outfit) spaces.
What if we were instead to do some more original things like giving it a custom `"disabled threshold"` of 0.05 (instead of the default 0.15)
It'd still be weak against plasma that'd overheat it but otherwise it should be able to gain 10% more effective hull without buffing its hull value too much. (It could makes sense lorewise caus there is scrap armor protecting it)
Or just give it more hull, that works as well. Would be nice to see this one as an actual tank I think.

**Comment by 1010todd** [comment]
> Why does it have less crew than the bastion? How would welding scrap metal onto it make it need less crew members?

Restructuring internal stuffs to support the extra mass. The description is a bit oversimplified.

**Comment by Zitchas** [comment]
I'm having serious doubts about these pirate ships. These appear to be just pushed through to have more pirate ships, without any explanation as to where they fit into the pirate ecosystem and how, exactly, we expect these ships to actually engage in piracy.

The Bastion, with it's big cargo hold, sounds like a ship that is tailor-made for piracy as-is. Good combination of combat capabilities and lots of cargo space to fill with loot. Then we get a piratified version... that loses all its cargo hold. Now it's just another warship, albeit one that conforms to the sort of view that purists have been arguing should be the default state of warships produced by formal navies: As in, they don't have cargo space (or at least very minimal), as cargo would be handled by fleets.

Now, if "Pirate", "Modified", or "Marauder" is basically shorthand for "Combat capable ships that retain enough cargo and passenger capability to keep solo players happy without needing a fleet..." I can get behind that. We need more jack-of-all-trade ships, and pirates are the very logical place to have ships oriented towards solo operations.

But these aren't being balanced for solo ship running, they're being balanced as if they're backed by large fleets with established logistics chains, and weirdly, they look *more* built to depend on a logistics chain than the *Navy* ships. These look more like dedicated combat ships. If these are criminals, they aren't pirates, they're assassins or thugs. 

But the main problem is that these are being thrown up (and merged) without much discussion as to why these ships, what role are they filling, how do they fit into the game? The most I've really heard is that Amazinite approved this list, so this list is going in.

**Comment by RisingLeaf** [comment]
How about opening a discussion where people can leave feedback and suggestions?
That way it is centralized in one place.

**Comment by Saugia** [comment]
Per recent events, I will be merging this PR once we have the respective copyright and asset/dpi PRs open, but we will *not* be implementing this ship in this PR, and I have removed but recorded the implementations of previous Pirate variant PRs. We will be discussing stats and balance in a future PR and/or a new discussion post before we implement these ships in fleets. Please bare with us until we get these ships in a more overall favorable state.

**Comment by Zitchas** [comment]
> This discussions _have_ happened in the discord many times and the PRs are left up for discussion and linked past the supposed 24 hour limit. Considering they _will_ be looked at and balanced in one large PR before the release of the next patch, I see no issue with merging them for now so they are all in one nice place to look at (different from HR which was merged and left to our devices for the next patch). That, and these ships are being opened by more than one person, and rather than inflate the PR count, we can merge them so they're all in conveniently one place.

Discord has had thousands of conversations on virtually every manner of topics, 99.9999% of which never amount to anything. There's been more than I can count on pirates. If there is existing discussion that is being implemented, than that should make it very easy for the initial PR to have the information necessary information to allow people to review the PR. We can't review a PR based on the entirety of conversation that takes places elsewhere, especially if it takes place places that people don't have access to. It could even be copy/pasted from the discussion (with suitable attribution), that would be acceptable.

> 
> These are not combat ships, and I disagree that their stats are representative of combat ships. I understand you are upset about the Clipper, but most ships are not following that trend and have respective pros and cons. Yes, they aren't balanced entirely, but that's going to be solved regardless.

Please don't make assumptions. I'm not upset. I'm a little frustrated that things are being merged without approvals or discussion. I would also point out that these are not for some isolated faction that are only being balanced internally or to future story; they are active in core human space, and so they have to be balanced against the 50+ ships already present in human space. All the other new pirate ships likewise must be balanced against all existing human ships, and right now, they're not meeting that bar.

> 
> While I agree that this many have been a poor practice, I do not appreciate you speaking about the problem in that way even though you were _part of several of those discussions_, and after I have addressed your concerns. I believe that to be an unfair representation of the situation, considering you were part these discussions, but are denying it, saying all you know is "Derpy this".

Please don't overstate my participation. I might have access to the private channel that I think a lot of this discussion took place in, and may have even posted at some point in part of this discussion, but that does not mean that I have a perfect knowledge of everything that has been said. Even if I did, as someone reviewing PRs, I have to review what in the PR, not what I assume might be done with it. 

> 
> In the future, perhaps it is better to merge the ships and such but not have them in fleets quite yet, as while they wouldn't be fully balanced, they wouldn't be in the game quite yet.

That would be good, yes. In the future, however, the best practice would be to provide a full explanation *in the PR* so that it can be evaluated by the reviewers. If these ships had been PR'd *without* insertion into fleets and stating up front that they were part of a set, and what the plan is for the set, and all those details; then I'd probably have been quite happy to just make a few comments and rubber-stamp it.

**Comment by Saugia** [comment]
>Discord has had thousands of conversations on virtually every manner of topics, 99.9999% of which never amount to anything. There's been more than I can count on pirates. If there is existing discussion that is being implemented, than that should make it very easy for the initial PR to have the information necessary information to allow people to review the PR. We can't review a PR based on the entirety of conversation that takes places elsewhere, especially if it takes place places that people don't have access to. It could even be copy/pasted from the discussion (with suitable attribution), that would be acceptable.

Again, as I keep saying, these PR's are less so much of large discussions on implementations, but more initial tests for these ships in vanilla during our time before the next patch release, and will be looked at in-depth in a future PR *before* we release it. While I say this may have not been the best approach, again, I believe you are far too exaggerating how important it is we do such an in-depth analysis now. The preference was to get these in sooner after some stat-work (which did happen), then gather everything together and do so *anyway* like I kept saying we will. It probably would have been better to implement these all at once, but unfortunately that's just not how it turned out. Regardless, either way, the discussion and stat work (has) will happen before the patch would be released.

>Please don't make assumptions. I'm not upset. I'm a little frustrated that things are being merged without approvals or discussion. I would also point out that these are not for some isolated faction that are only being balanced internally or to future story; they are active in core human space, and so they have to be balanced against the 50+ ships already present in human space. All the other new pirate ships likewise must be balanced against all existing human ships, and right now, they're not meeting that bar.

If you would like assumptions to not be made, then please do so in return. We actually did decent work on some of the previous pirate ships, and a lot of people were happy with them. To say they aren't meeting the bar comes off as more of a personal opinion of yours and most certainly not an overall basis. so don't assume that opinion is the end-all.

>Please don't overstate my participation. I might have access to the private channel that I think a lot of this discussion took place in, and may have even posted at some point in part of this discussion, but that does not mean that I have a perfect knowledge of everything that has been said. Even if I did, as someone reviewing PRs, I have to review what in the PR, not what I assume might be done with it.

Then please don't understate it by saying "I only know Derpy said this," because that isn't the truth.

>That would be good, yes. In the future, however, the best practice would be to provide a full explanation in the PR so that it can be evaluated by the reviewers. If these ships had been PR'd without insertion into fleets and stating up front that they were part of a set, and what the plan is for the set, and all those details; then I'd probably have been quite happy to just make a few comments and rubber-stamp it.

Preferably they would be introduced all in one PR, or simply the artwork. Again, this happened in a way that wasn't the best and I acknowledge that, so I'm fixing it by merging what we have and removing the fleet implementation until we get to balance. Let's leave it at that.

**Comment by Zitchas** [comment]
Thank you for removing the inclusion in fleets, and stating that implementation and balance will be done in another PR shortly.

And yes, of course it's my personal opinion. There's no meter stick we can pull out to objectively measure it with, so *all* reviews, if they aren't purely just correcting typos or mistakes, are personal opinions. 

That being said, I'm expected to review new content  based on the lore/theme and how they fit into the existing lore/setting. It's my *responsibility* to find out what the lore is for these ships, and if there is insufficient lore and/or it doesn't appear to fit well within our setting, to request changes. So please don't take it personally if I question the lack of thereof.

Agreed, let us learn from this experience and move forward.

**Comment by 1010todd** [comment]
> I'm having serious doubts about these pirate ships. These appear to be just pushed through to have more pirate ships, without any explanation as to where they fit into the pirate ecosystem and how, exactly, we expect these ships to actually engage in piracy.

It's in the ship's description, though, seems like the edits might've clouded that a bit. Anyway, the Bulwark by Lake's request is a pure tough gun platform operated by larger groups to protect their establishments. This one specifically is not for operating solo but as part of defensive fleet against raids/attack by either the government like Republic (or most likely in current situation, Free Worlds) or other rival gangs who may possess enough firepower to be a threat to the operators of Bulwarks. So according to Lake this will most often be part of planetary defense fleet and should be rare to see else where. So yeah, this ship would be less of a pirate captain or a small group of them but more often used by one of the more organized groups, say the the group(s) that control Greenrock for example would be the most likely operator.

For other options they could've used like Falcon or just regular Bastion, both have rather high cargo capacity and might've been preferred to let them go out, plunder, and make money for them while Bulwark will always be at the planet guarding it against any surprise attack and hold out until reinforcement comes and hopefully even repel the attack by themselves. If it were a normal Bastion or Falcon who ever owns them is wasting hundreds of tons of cargo they could've been plundering for a standard warship anyone could get their hands on while Bulwark have majority of the cargo space they don't need converted into additional space and reinforcement for the ship and make it excels at the job of just beating other ships. And yeah, this will require some logistic backing and may even require more than Navy ships but these shouldn't travel too far from their home base in majority of the case.

**Comment by Saugia** [comment]
Per recent events, I will be merging this PR, but we will not be implementing this ship in this PR, and I have removed but recorded the implementations of previous Pirate variant PRs. We will be discussing stats and balance in a future PR and/or a new discussion post before we implement these ships in fleets. Please bare with us until we get these ships in a more overall favorable state.

**Pull request merged**